src/Typed/Resources/assets/src/controller.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+import { Controller } from 'stimulus';
+import Typed from "typed.js";
+
+export default class extends Controller {
+
+    static values = {
+        strings: Array,
+        typeSpeed: Number,
+        smartBackspace: Boolean,
+        startDelay: Number,
+        backSpeed: Number,
+        shuffle: Boolean,
+        backDelay: Number,
+        fadeOut: Boolean,
+        fadeOutClass: String,
+        fadeOutDelay: Number,
+        loop: Boolean,
+        loopCount: Number,
+        showCursor: Boolean,
+        cursorChar: String,
+        autoInsertCss: Boolean,
+        attr: String,
+        bindInputFocusEvents: Boolean,
+        contentType: String,
+    }
+
+    connect() {
+        const options = {
+            strings: this.stringsValue || null,
+            typeSpeed: this.typeSpeedValue || 30,
+            smartBackspace: this.smartBackspaceValue || true,
+            startDelay: this.startDelayValue || 0,
+            backSpeed: this.backSpeedValue || 0,
+            shuffle: this.shuffleValue || false,
+            backDelay: this.backDelayValue || 700,
+            fadeOut: this.fadeOutValue || false,
+            fadeOutClass: this.fadeOutClassValue || 'typed-fade-out',
+            fadeOutDelay: this.fadeOutDelayValue || 500,
+            loop: this.loopValue || false,
+            loopCount: this.loopCountValue || Infinity,
+            showCursor: this.showCursorValue || true,
+            cursorChar: this.cursorCharValue || '|',
+            autoInsertCss: this.autoInsertCssValue || true,
+            attr: this.attrValue || null,
+            bindInputFocusEvents: this.bindInputFocusEventsValue || false,
+            contentType: this.contentTypeValue || 'html',
+            onBegin: this.onBegin,
+            onComplete: this.onComplete,
+            preStringTyped: this.preStringTyped,
+            onStringTyped: this.onStringTyped,
+            onLastStringBackspaced: this.onLastStringBackspaced,
+            onTypingPaused: this.onTypingPaused,
+            onTypingResumed: this.onTypingResumed,
+            onReset: this.onReset,
+            onStop: this.onStop,
+            onStart: this.onStart,
+            onDestroy: this.onDestroy,
```

Comment:
All of these options that are callbacks... I don't think it's possible to pass a callback in as a Stimulus value (and, indeed, you have not tried to fetch these from the values system). So what is the purpose of these?

Reply:
Yes these are callbacks. It looks like you are right: I was not able to use these methods from outside the component.
I remove those lines for now.